tools/clang/tools/dxcompiler/dxcshadersourceinfo.h

```diff
@@ -43,19 +32,27 @@ struct SourceInfoReader {
     llvm::StringRef Content;
   };
 
+  struct ArgPair {
+    std::string Name;
+    std::string Value;
+    inline std::string Render() const {
+      if (Name.size())
+        return std::string("-") + Name + Value;
```

Comment:
I think this assumes that all flags with values support joined form.  I don't think they do.  But I think that all flags we use do support separate form.  But if separate, I don't think you want to join these together into one output from Render() in the first place, right?  So there are implications to the code that calls this.